src/utils/generateUID.js

```diff
@@ -0,0 +1,22 @@
+export const generateUID = (prefix, separator) => {
+  let counter = 1
+
+  let map = new WeakMap()
+
+  const get = item => {
+    if (!map.has(item)) {
+      map.set(item, counter++)
+      return get(item)
+    }
+    return `${prefix || 'uid'}${separator}${map.get(item)}`
+  }
+
+  const reset = () => {
+    map = new WeakMap()
+    counter = 1
+  }
+
+  return { get, reset }
+}
+
+export const rdtsUID= generateUID('rdts', '')
```

Comment:
Unless we plan to make it a generic utility method usable outside of the component, I guess we can tighten this up a bit more.

E.g.
 - `generateUID` is not used anywhere outside of this file, so we can choose not to export it
 - eliminate branches like `prefix || 'uid'`
 - eliminate optional separator logic - decide if we want to have a separator or not and then design the API accordingly.

Next, about its API surface; imagine someone **new** looking at `this.clientId = props.id || rdtsUID.get(this)` and trying to understand the code. 

Is `rdtsUID.get()` self-descriptive enough or can we do more - so that one doesn't need to peek at its implementation to figure out what it means or does? How about something like `getInstanceId.next(this)` or `nextUniqueId.get(this)` or `clientIdGenerator.get(this)` or similar? 

If one is curious about _how_ it's implemented, they would still need to peek at the implementation, but if I'm just trying to understand the overall flow, maybe this tells me enough about what the intended op is?

Reply:
I agreed/landed with clientIdGenerator